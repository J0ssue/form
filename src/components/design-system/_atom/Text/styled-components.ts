import styled from "@emotion/styled";
import { css } from "@emotion/react";

export const TextComp = styled.p<{
  color: string;
  size: number;
  align: string;
  transform: string;
  lineHeight?: number;
}>`
  ${({ color, size, transform, align, lineHeight }) => css`
    color: ${color};
    font-size: ${size}px;
    text-align: ${align};
    text-transform: ${transform};
    line-height: ${lineHeight || "normal"}px;
  `}
`;
