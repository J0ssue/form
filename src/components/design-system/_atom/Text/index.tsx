import React from "react";
import buildColor from "../../_static/colorPalette";

import { TextComp } from "./styled-components";

interface Props {
  /**
   * defines element tag
   * */
  tag: React.ElementType<string> | undefined;
  /**
   * defines color
   * */
  color: string;
  /**
   * defines text-align
   * */
  align: string;
  /**
   * defines font-size
   * */
  size: number;
  /**
   * defines text-transform
   * */
  transform: string;
  /**
   * defines line-height
   * */
  lineHeight?: number;
  /**
   * inserts text
   * */
  text?: string;
}

const Text = (props: Props): JSX.Element => {
  return (
    <TextComp {...props} as={props.tag}>
      {props.text || ""}
    </TextComp>
  );
};

Text.defaultProps = {
  color: buildColor("black", "900"),
  size: 12,
  transform: "none",
  tag: "p",
  align: "left",
};

export default Text;
