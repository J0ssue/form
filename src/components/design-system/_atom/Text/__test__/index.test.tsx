// import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Text from "..";

describe("<Text/> rendering", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Text text="hello world" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
