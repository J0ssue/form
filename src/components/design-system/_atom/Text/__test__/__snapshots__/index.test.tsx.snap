// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text/> rendering should render correctly 1`] = `
<p
  className="css-fp2tty"
  color="#000000"
  size={12}
  transform="none"
>
  hello world
</p>
`;
